codecov.yml

```diff
@@ -2,6 +2,8 @@ codecov:
   require_ci_to_pass: yes
   notify:
     after_n_builds: 3
+  ignore:
```

Comment:
I think it's better to remove this file, because in future if the file get new code added to it, then it may be missed from coverage or no need to ignore it

Reply:
We don't add code in `__init__.py` as it's bad practice. It should only be used for exporting modules to a package's public interface. We can't remove the file as the packaging system requires it.

Reply:
Ok, there are some many `__init__.py` which can be part of this commit., to improve Code coverage

Reply:
Example : `src\mbed_tools\lib\__init__.py` will this be included to?

Reply:
`**/*` is a recursive glob in an attempt to ignore all of them. Although looking at the report it's not ignoring `templates/__init__.py` for some reason, I'll see if that's my fault or codecov's.

Reply:
Reverted this change as it wasn't having the desired effect.

Reply:
Might be worth adding an individual entry for each file. I don't think there are a lot of them, which may also improve code readability to figure out "where to start"

Reply:
I don't think it's worth it for "code readability". If I was looking for "where to start" I'd rather see a code style guide. It's kinda unexpected to study the codecov config file for stuff like that. We should have a style guide, though. I will add a task for it to the backlog.

Reply:
Added a task for code style guide https://jira.arm.com/browse/IOTBTOOL-957